### PR TITLE
Update Gem::Package call and remove vendored version

### PR DESCRIPTION
**Pull request by arthurnn**
[continuation of #686]

Right now we are running under rubygems 2.2.2 (same version we forced on travis)
When we upgrade, we vendored the `Package` class to keep the old calls on rubygems.org (see https://github.com/rubygems/rubygems.org/commit/82b3dc2c)

This removed the vendor so we can use main rubygem again, and fix the call.

review @drbrain @evanphx @dwradcliffe @qrush 

As far as I can tell, the old PR were approved but never merged.

### CODE REVIEW

**Comment by arthurnn** [comment]
I rebased this, and this should test under latest rubygems and 2.2.2 rubygems which is the one which we use in production.

**Comment by dwradcliffe** [comment]
I think this makes sense... no reason to keep old code around. :+1:

File: app/models/pusher.rb [hunk 1]
```diff
@@ -44,12 +42,15 @@ def save
   end
 
   def pull_spec
-    GemPackage.open body, "r", nil do |pkg|
-      @spec = pkg.metadata
-      return true
+    gem_tar = Gem::Package::TarReader.new body
```
Comment (drbrain) [inline app/models/pusher.rb]: This should switch to `Gem::Package.new(body).load_spec` when rubygems.org is upgraded to RubyGems 2.3 or newer, can we leave a comment?

Comment (arthurnn) [inline app/models/pusher.rb]: good catch! I will add a comment.

**Comment by arthurnn** [comment]
@drbrain  Comment added.

also cc, @xaviershay as he added support for receiving a IO as package body on https://github.com/rubygems/rubygems/pull/716 , which we will use after we upgrade our rubygems version.

**Comment by drbrain** [comment]
My name is Eric Hodel and I approve this pull request.

**Pull request merged**